components/forms/CreateTxForm/MsgForm/MsgTransferForm.tsx

```diff
@@ -0,0 +1,233 @@
+      if (foundTimeout) {
+        return Long.fromNumber(Date.now() + foundTimeout).multiply(1_000_000); // In nanoseconds
+      }
+
+      try {
+        return Long.fromString(timeout);
+      } catch {
+        return Long.fromNumber(0);
+      }
+    })();
+
+    const msg: TxMsgTransfer = {
+      typeUrl: "/ibc.applications.transfer.v1.MsgTransfer",
+      value: {
+        sender: fromAddress,
+        receiver: toAddress,
+        token: { denom, amount },
+        sourcePort,
+        sourceChannel,
+        timeoutTimestamp,
+        memo,
+      },
+    };
+
+    setMsgGetter({ isMsgValid, msg });
+  }, [
+    amount,
+    denom,
+    fromAddress,
+    memo,
+    setMsgGetter,
+    sourceChannel,
+    sourcePort,
+    state.chain.chainId,
+    timeout,
+    toAddress,
+  ]);
+
+  return (
+    <StackableContainer lessPadding lessMargin>
+      <button className="remove" onClick={() => deleteMsg()}>
+        ✕
+      </button>
+      <h2>MsgTransfer</h2>
+      <div className="form-item">
+        <Input
+          label="Recipient Address"
+          name="recipient-address"
+          value={toAddress}
+          onChange={({ target }) => setToAddress(target.value)}
+          error={toAddressError}
+          placeholder={`E.g. ${exampleAddress(0, state.chain.addressPrefix)}`}
+        />
+      </div>
+      <div className="form-item">
+        <Input
+          label="Denom"
+          name="denom"
+          value={denom}
+          onChange={({ target }) => setDenom(target.value)}
+          error={denomError}
+        />
+      </div>
+      <div className="form-item">
+        <Input
+          type="number"
+          label="Amount"
+          name="amount"
+          value={amount}
+          onChange={({ target }) => setAmount(target.value)}
+          error={amountError}
+        />
+      </div>
+      <div className="form-item">
+        <Input
+          label="Source Port"
+          name="source-port"
+          value={sourcePort}
+          onChange={({ target }) => setSourcePort(target.value)}
+          error={sourcePortError}
+        />
+      </div>
+      <div className="form-item">
+        <Input
+          label="Source Channel"
+          name="source-channel"
+          value={sourceChannel}
+          onChange={({ target }) => setSourceChannel(target.value)}
+          error={sourceChannelError}
+        />
+      </div>
+      <div className="form-item">
+        <Input
+          type="text"
+          list="timestamp-options"
+          label="Timeout"
+          name="timeout"
+          placeholder="Enter timestamp in nanoseconds or select from list"
+          value={timeout}
+          onChange={({ target }) => setTimeout(target.value)}
+          error={timeoutError}
+        />
+        <datalist id="timestamp-options">
+          {humanTimestampOptions.map(({ label }) => (
+            <option key={label}>{label}</option>
+          ))}
+        </datalist>
```

Comment:
I tried this dropdown in the Vercel preview. But it seems like after it was set once you cannot change it anymore.

Reply:
Ah it seems it filters the list, maybe I can clear the field on click so all the options appear